sympy/stats/crv.py

```diff
@@ -174,76 +171,78 @@ class SampleContinuousNumpy:
     def __new__(cls, dist, size):
         return cls._sample_numpy(dist, size)
 
-    numpy_rv_map = {
-        'BetaDistribution': lambda dist, size: numpy.random.beta(a=float(dist.alpha),
-            b=float(dist.beta), size=size),
-        'ChiSquaredDistribution': lambda dist, size: numpy.random.chisquare(
-            df=float(dist.k), size=size),
-        'ExponentialDistribution': lambda dist, size: numpy.random.exponential(
-            1/float(dist.rate), size=size),
-        'GammaDistribution': lambda dist, size: numpy.random.gamma(float(dist.k),
-            float(dist.theta), size=size),
-        'LogNormalDistribution': lambda dist, size: numpy.random.lognormal(
-            float(dist.mean), float(dist.std), size=size),
-        'NormalDistribution': lambda dist, size: numpy.random.normal(
-            float(dist.mean), float(dist.std), size=size),
-        'ParetoDistribution': lambda dist, size: (numpy.random.pareto(
-            a=float(dist.alpha), size=size) + 1) * float(dist.xm),
-        'UniformDistribution': lambda dist, size: numpy.random.uniform(
-            low=float(dist.left), high=float(dist.right), size=size)
-    }
-
     @classmethod
     def _sample_numpy(cls, dist, size):
         """Sample from NumPy."""
 
-        dist_list = cls.numpy_rv_map.keys()
+        import numpy
+        numpy_rv_map = {
+            'BetaDistribution': lambda dist, size: numpy.random.beta(a=float(dist.alpha),
+                b=float(dist.beta), size=size),
+            'ChiSquaredDistribution': lambda dist, size: numpy.random.chisquare(
+                df=float(dist.k), size=size),
+            'ExponentialDistribution': lambda dist, size: numpy.random.exponential(
+                1/float(dist.rate), size=size),
+            'GammaDistribution': lambda dist, size: numpy.random.gamma(float(dist.k),
+                float(dist.theta), size=size),
+            'LogNormalDistribution': lambda dist, size: numpy.random.lognormal(
+                float(dist.mean), float(dist.std), size=size),
+            'NormalDistribution': lambda dist, size: numpy.random.normal(
+                float(dist.mean), float(dist.std), size=size),
+            'ParetoDistribution': lambda dist, size: (numpy.random.pareto(
+                a=float(dist.alpha), size=size) + 1) * float(dist.xm),
+            'UniformDistribution': lambda dist, size: numpy.random.uniform(
+                low=float(dist.left), high=float(dist.right), size=size)
+        }
+
+        dist_list = numpy_rv_map.keys()
 
         if dist.__class__.__name__ not in dist_list:
             return None
 
-        return cls.numpy_rv_map[dist.__class__.__name__](dist, size)
+        return numpy_rv_map[dist.__class__.__name__](dist, size)
```

Comment:
While you're here, this simplifies to
```python
return numpy_rv_map.get(dist.__class__.__name__, lambda dist, size: None)(dist, size)
```
or if that feels ugly
```python
f = numpy_rv_map.get(dist.__class__.__name__)
if f is None:
    return None
return f(dist, size)
```

Reply:
Although I suppose there are a lot of these, so fine to leave to another PR.

Reply:
I would leave this for another PR. We should use subclasses rather than dictionaries.